data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -136,66 +146,124 @@ private void parseFile(String filePath){
 
     @Override
     public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
-        //todo
-        for(final Record<Event> record : records) {
+        for (final Record<Event> record : records) {
             final Event recordEvent = record.getData();
-            if (Objects.nonNull(translateProcessorConfig.getMapWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getMapWhen(), recordEvent)) {
+            if (Objects.nonNull(translateProcessorConfig.getTranslateWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getTranslateWhen(), recordEvent)) {
                 continue;
             }
             try {
-                String matchKey = record.getData().get(translateProcessorConfig.getSource(), String.class);
-                if(matchesIndividualEntry(record, matchKey) || matchesRangeEntry(record, matchKey) || matchesPatternEntry(record, matchKey)){
-                    continue;
+                if (Objects.nonNull(translateProcessorConfig.getIterateOn())) {
+                    List<Map<String, Object>> nestedObjects = recordEvent.get(translateProcessorConfig.getIterateOn(), List.class);
+                    for (Map<String, Object> nextedObject : nestedObjects) {
+                        performMappings(nextedObject);
+                    }
+                    recordEvent.put(translateProcessorConfig.getIterateOn(), nestedObjects);
+                } else {
+                    performMappings(recordEvent);
                 }
-                else{
+            } catch (Exception ex) {
+                LOG.error(EVENT, "Error mapping the source [{}] of entry [{}]", translateProcessorConfig.getSource(),
+                          record.getData(), ex);
+            }
+        }
+        return records;
+    }
 
-                    // todo : add default, increment metrics, and/or add_tags
+    private String getSourceValue(Object nestedObject, String sourceKey) {
+        if (nestedObject instanceof Map) {
```

Comment:
`performMappings()`calls `getSourceValue()`
It will be true when `performMappings` is called on the nested Json object (which is a Map<String, Object>).
`performMappings` when called with an Object of type Event, it will be false.
I think I need to rename the getSourceValue parameters for better understanding.